test/Microsoft.Azure.WebJobs.Extensions.Kafka.LangEndToEndTests/Common/Queue/AzureStorageQueueManager.cs

```diff
@@ -0,0 +1,80 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Threading.Tasks;
+using Azure.Storage.Queues;
+using Azure.Storage.Queues.Models;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.WebJobs.Extensions.Kafka.LangEndToEndTests.Common
+{
+	// Implementation of IQueueManager responsible for management of Azure Queue Resource.
+	public class AzureStorageQueueManager : IQueueManager<QueueRequest, QueueResponse>
+	{
+		private readonly string _connectionString;
+		private static readonly AzureStorageQueueManager _instance = new ();
+		private readonly ConcurrentDictionary<string, QueueClient> _queueClientStore;
+		private readonly ILogger _logger = TestLogger.GetTestLogger();
+
+		public static AzureStorageQueueManager GetInstance()
+		{
+			return _instance;
+		}
+
+		private AzureStorageQueueManager()
+		{
+			_connectionString = Environment.GetEnvironmentVariable(Constants.AZURE_WEBJOBS_STORAGE);
+			_queueClientStore = new ConcurrentDictionary<string, QueueClient>();
+		}
+
+		public async Task ClearAsync(string queueName)
+		{
+			QueueClient queueClient = new QueueClient(_connectionString, queueName);
```

Comment:
Shouldnt the queue always exist in this function. Is it optimized to check if the queue client is cached or not?

Reply:
Updated to a cleaner implementation.